### PR TITLE
G38 bug fix, made double touch optional & reverted to Z_MIN_PROBE

### PR DESCRIPTION
**Pull request by Bob-the-Kuhn**
Made the double touch portion a conditional compile based on the PROBE_DOUBLE_TOUCH flag.

### CODE REVIEW

**Comment by Bob-the-Kuhn** [comment]
This PR is ready for review.

**Comment by Bob-the-Kuhn** [comment]
The current G38 only stopped a move if it involved the Z axis.

Moved all the G38 code to its own section and put it where it would always be executed no matter what axis was moving or if the endstop was enabled.

Also added a comment to configuration_adv to alert the user the double tap had to be turned on.

The bug was discovered in issue #5743.

**Comment by Bob-the-Kuhn** [comment]
Just added another change to G38.  This one changes the pin being used from Z_MIN to Z_MIN_PROBE.

We originally started out with Z_MIN_PROBE.  I don't remember why we changed to Z_MIN endstop before it was merged.

In my config there is no Z_MIN_ endstop.  It's being used as one of the Z_DUAL_ENDSTOPS MAX.

**Comment by Phife54** [comment]
Awesome Bob, Thanks for this. Only Question, you mentioned to me that this only works in the -Z direction, can this be changed to work in the +Z direction but still use the Z Probe Pin?

Z Min endstop is my home switch, probing takes place when moving away from home. probe is on X_MAX and is defined in configuration.h as #define Z_MIN_PROBE_PIN X_MAX_PIN

File: Marlin/endstops.cpp [hunk 1]
```diff
@@ -246,26 +246,28 @@ void Endstops::update() {
...
       UPDATE_ENDSTOP_BIT(AXIS, MINMAX); \
       if (TEST_ENDSTOP(_ENDSTOP(AXIS, MINMAX)) && stepper.current_block->steps[_AXIS(AXIS)] > 0) { \
         _ENDSTOP_HIT(AXIS); \
         stepper.endstop_triggered(_AXIS(AXIS)); \
-        CODE; \
       } \
     } while(0)
 
-  #if ENABLED(G38_PROBE_TARGET) && PIN_EXISTS(Z_MIN)  // If G38 command then check Z_MIN for every axis and every direction
+  #define UPDATE_ENDSTOP(AXIS,MINMAX) _UPDATE_ENDSTOP(AXIS,MINMAX)
```
Comment (thinkyhead) [inline Marlin/endstops.cpp]: Just define the above macro as `UPDATE_ENDSTOP` and drop this.

File: Marlin/Configuration_adv.h [hunk 2]
```diff
@@ -578,6 +578,7 @@
 //#define BEZIER_CURVE_SUPPORT
 
 // G38.2 and G38.3 Probe Target
+// enable PROBE_DOUBLE_TOUCH if you want G38 to double touch
```
Comment (thinkyhead) [inline Marlin/Configuration_adv.h]: Capitalize "Enable" here. And add this line to all other `Configuration_adv.h` files.
Search:
```
// G38.2 and G38.3 Probe Target
//#define G38_PROBE_TARGET
```
Replace:
```
// G38.2 and G38.3 Probe Target
// Enable PROBE_DOUBLE_TOUCH if you want G38 to double touch
//#define G38_PROBE_TARGET
```

**Comment by thinkyhead** [comment]
Overall, good. Small tweaks and it's ready to merge.

**Comment by Bob-the-Kuhn** [comment]
All changes have been implemented, the example config_adv files have been updated and it's been squashed to one commit.

This PR is ready for prime time.

**Comment by Bob-the-Kuhn** [comment]
Just rebased it.

**Comment by thinkyhead** [comment]
Working my way backward through a pile of ~300 issues. Finally arrived at this one!

**Pull request merged**